File: programmers/난이도별/level02.롤케이크_자르기/SSSOy.py
def solution(topping):
    answer = 0
    topping_set = set()
    topping_dict = {}
    
    for i in topping:
        str_i = str(i)
        topping_dict[str_i] = topping_dict.get(str_i, 0)
        topping_dict[str_i] += 1
        
    for i in topping:
        
        topping_set.add(i)
        
        str_i = str(i)
        topping_dict[str_i] -= 1
        if topping_dict[str_i] <= 0:
            del(topping_dict[str_i])
        
        set_len = len(topping_set)
        dict_len = len(topping_dict)
        
        if set_len > dict_len:
            break
        elif set_len == dict_len:
            answer += 1
    
    return answer

File: programmers/난이도별/level01_바탕화면_정리/SSSOy.java
class Solution {
    public int[] solution(String[] wallpaper) {
        int[] answer = new int[] { 50, 50, 0, 0 };
        
        for (int i = 0; i < wallpaper.length; i++) {
            char[] files = wallpaper[i].toCharArray();
            
            for (int j = 0; j < files.length; j++) {
                if (files[j] == '#') {
                    
                    answer[0] = answer[0] > i? i : answer[0];
                    answer[1] = answer[1] > j? j : answer[1];
                    answer[2] = answer[2] < i + 1? i + 1 : answer[2];
                    answer[3] = answer[3] < j + 1? j + 1 : answer[3];
                }
            }
        }
        
        return answer;
    }
}
